src/main/java/seedu/address/logic/parser/RemoveTagCommandParser.java

```diff
@@ -20,29 +22,34 @@
      * @throws ParseException if the user input does not conform the expected format
      */
     public RemoveTagCommand parse(String args) throws ParseException {
-        Index index = null;
+        requireNonNull(args);
+
         String trimmedArgs = args.trim();
+        String[] fields = trimmedArgs.split("\\s+");
+
+        Index index;
+        Tag toRemove;
 
-        if (trimmedArgs.isEmpty()) {
+        if (trimmedArgs.isEmpty() || fields.length < 2) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, RemoveTagCommand.MESSAGE_USAGE));
         }
 
-        if (trimmedArgs.split(" ").length >= 3) {
+        if (fields.length > 2) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, RemoveTagCommand.MESSAGE_EXCEEDTAGNUM));
         }
 
         try {
-            if (trimmedArgs.contains(" ")) {
-                index = ParserUtil.parseIndex(trimmedArgs.split(" ")[0]);
-                trimmedArgs = trimmedArgs.split(" ")[1];
-            }
-            Tag toRemoved = new Tag(trimmedArgs);
-            if (index == null) {
-                return new RemoveTagCommand(toRemoved);
+            toRemove =  new Tag(fields[RemoveTagCommand.TAG]);
+            if (fields[RemoveTagCommand.RANGE].equals(RemoveTagCommand.WHOLEAB)) {
+                return new RemoveTagCommand(toRemove);
+            } else if (StringUtil.isNonZeroUnsignedInteger(fields[RemoveTagCommand.RANGE])) {
+                index = ParserUtil.parseIndex(fields[RemoveTagCommand.RANGE]);
```

Comment:
Three constants declared in the `RemoveTagCommand` class used only in `RemoveTagCommandParser` class. As mentioned above, I recommend moving all three into this class instead.